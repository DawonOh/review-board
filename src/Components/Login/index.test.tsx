import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import 'jest-styled-components';
import { Login } from '.';
import { MemoryRouter as Router } from 'react-router-dom';

test('Login render correctly', () => {
  let isModalOpen = true;
  const setIsModalOpen = (props: boolean) => {
    isModalOpen = props;
  };
  const { container } = render(
    <Router>
      <Login isModalOpen={isModalOpen} setIsModalOpen={setIsModalOpen} />
    </Router>
  );
  const loginModal = screen.getByTestId('Login Test');
  expect(loginModal).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

test('enable button when email and password are entered', () => {
  let isModalOpen = true;
  const setIsModalOpen = (props: boolean) => {
    isModalOpen = props;
  };
  render(
    <Router>
      <Login isModalOpen={isModalOpen} setIsModalOpen={setIsModalOpen} />
    </Router>
  );
  const emailInput = screen.getByPlaceholderText('이메일을 입력해주세요.');
  const passwordInput = screen.getByPlaceholderText('비밀번호를 입력해주세요.');
  const button = screen.getByRole('button');

  expect(button).toBeDisabled();

  fireEvent.change(emailInput, { target: { value: 'example@email.com' } });
  fireEvent.change(passwordInput, { target: { value: 'test1234!' } });

  expect(button).toBeEnabled();
});

test('can not login when button is disabled', () => {
  const login = jest.fn();
  let isModalOpen = true;
  const setIsModalOpen = (props: boolean) => {
    isModalOpen = props;
  };
  render(
    <Router>
      <Login isModalOpen={isModalOpen} setIsModalOpen={setIsModalOpen} />
    </Router>
  );

  const button = screen.getByRole('button');
  fireEvent.click(button);
  expect(login).toHaveBeenCalledTimes(0);
});
