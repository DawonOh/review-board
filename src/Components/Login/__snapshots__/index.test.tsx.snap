// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 30;
  background-color: rgba(0,0,0,0.5);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  padding: 0.6em 1em;
  background-color: #E0E0E0;
  font-size: 1em;
  border: none;
  border-radius: 0.3em;
  cursor: default;
}

.c8 {
  color: #fff;
}

.c2 {
  width: 100%;
  height: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.2rem;
}

.c3 {
  position: absolute;
  top: 2rem;
  right: 2rem;
  width: 1rem;
  height: 1rem;
  background: url(close.png);
  background-repeat: no-repeat;
  background-size: cover;
  cursor: pointer;
}

.c4 {
  font-size: 2em;
  font-weight: 700;
  color: #676FA3;
}

.c5 {
  width: 20em;
  padding: 0.6em;
  font-size: 1em;
  border: 1px solid #e0e0e0;
  border-radius: 0.3em;
}

.c5:focus {
  outline: none;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.3rem;
}

.c9 {
  font-size: 0.9em;
}

.c1 {
  width: 32em;
  height: 20em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: white;
  border-radius: 4px;
}

<div>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
        data-testid="Login Test"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
            data-testid="close button"
          />
          <h1
            class="c4"
          >
            Title
          </h1>
          <input
            class="c5"
            placeholder="이메일을 입력해주세요."
            type="text"
            value=""
          />
          <input
            class="c5"
            placeholder="비밀번호를 입력해주세요."
            type="password"
            value=""
          />
          <div
            class="c6"
          >
            <button
              class="c7"
              color="#fff"
              data-testid="container-element"
              disabled=""
            >
              <span
                class="c8"
                color="#fff"
              >
                로그인
              </span>
            </button>
          </div>
          <a
            href="/join"
          >
            <span
              class="c9"
            >
              회원가입
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
